clippy_lints/src/dbg_macro.rs

```diff
@@ -0,0 +1,55 @@
+use crate::utils::span_lint_and_sugg;
+use rustc::lint::{EarlyContext, EarlyLintPass, LintArray, LintPass};
+use rustc::{declare_tool_lint, lint_array};
+use rustc_errors::Applicability;
+use syntax::ast;
+
+/// **What it does:** Checks for usage of dbg!() macro.
+///
+/// **Why is this bad?** `dbg!` macro is intended as a debugging tool. It
+/// should not be in version control.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust,ignore
+/// // Bad
+/// dbg!(true)
+///
+/// // Good
+/// true
+/// ```
+declare_clippy_lint! {
+    pub DBG_MACRO,
+    restriction,
+    "`dbg!` macro is intended as a debugging tool"
+}
+
+#[derive(Copy, Clone, Debug)]
+pub struct Pass;
+
+impl LintPass for Pass {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(DBG_MACRO)
+    }
+
+    fn name(&self) -> &'static str {
+        "DbgMacro"
+    }
+}
+
+impl EarlyLintPass for Pass {
+    fn check_mac(&mut self, cx: &EarlyContext<'_>, mac: &ast::Mac) {
+        if mac.node.path == "dbg" {
+            span_lint_and_sugg(
+                cx,
+                DBG_MACRO,
+                mac.span,
+                "`dbg!` macro is intended as a debugging tool",
+                "ensure to avoid having uses of it in version control",
+                mac.node.tts.to_string(),
```

Comment:
Heh, I think we can just nuke the entire invocation (so `String::new()` here). Also this suggestion would make `dbg!("{:?}", foo);` become `"{:?}", foo;` which is not great Rust code

Reply:
`dbg!()` can only one argument. I guess you're misunderstanding something.

https://github.com/rust-lang/rust/blob/167ceff01ec2f01f677fa6351646255d3dacbb98/src/libstd/macros.rs#L311

Reply:
@oli-obk I believe you are missing the context here. `dbg!(_)` is meant as something you can easily add to your code, see this example https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=5e656cc65bf3e8296ef299960f2fa51f

Reply:
Indeed. Don't mind me then xD.